chain-core/src/state/account/op/data/attribute.rs

```diff
@@ -0,0 +1,42 @@
+use parity_scale_codec::{Decode, Encode, Error, Input, Output};
+#[cfg(not(feature = "mesalock_sgx"))]
+use serde::{Deserialize, Serialize};
+
+/// attributes in StakedState-related transactions
+#[derive(Debug, Default, PartialEq, Eq, Clone)]
+#[cfg_attr(not(feature = "mesalock_sgx"), derive(Serialize, Deserialize))]
+pub struct StakedStateOpAttributes {
+    pub chain_hex_id: u8,
+    pub app_version: u64,
+}
+
+impl StakedStateOpAttributes {
+    pub fn new(chain_hex_id: u8) -> Self {
+        StakedStateOpAttributes {
+            chain_hex_id,
+            app_version: crate::APP_VERSION,
+        }
+    }
+}
+
+impl Encode for StakedStateOpAttributes {
+    fn encode_to<EncOut: Output>(&self, dest: &mut EncOut) {
+        dest.push_byte(self.chain_hex_id);
+        dest.push(&self.app_version);
+    }
+
+    fn size_hint(&self) -> usize {
+        1 + self.chain_hex_id.size_hint() + self.app_version.size_hint()
```

Comment:
Why to add `1` here? We're only serializing `chain_hex_id` and `app_version`.

Reply:
it's now adding an extra byte as well

Reply:
A one byte tag future extension.

Reply:
Shouldn’t we write a default tag value (`0`) and verify when decoding?